app/editor_ui/mixins.py

```diff
@@ -20,3 +23,99 @@ def form_valid(self, form):
         instance.created_by = self.request.user
 
         return super().form_valid(form)
+
+
+class ProjectOwnerMembershipMixin:
+    """
+    Mixin to automatically create a ProjectMembership for the user as the project owner
+    when a Project is created (unless the user is a superuser).
+    """
+
+    def form_valid(self, form):
+        response = super().form_valid(form)
+        user = self.request.user
+        instance = self.object
+
+        if isinstance(instance, Project) and not user.is_superuser:
+            ProjectMembership.objects.get_or_create(
+                user=user,
+                project=instance,
+                defaults={"role": "owner"},
+            )
+
+        return response
+
+
+class ProjectMembershipRequiredMixin:
+    """
+    Mixin to restrict access to views based on project membership and required roles.
+
+    Uses the `project_relation` attribute to determine the ForeignKey path to Project.
+    For CreateView/ListView, expects 'project_uuid' in URL kwargs.
+    Set `required_project_roles` as a class attribute (e.g., ["editor"]).
+    """
+
+    required_project_roles = None
+    project_url_kwarg = "project_uuid"
+
+    def get_project_instance(self, obj):
+        """
+        Given an object (e.g., FeedbackForm), follow ForeignKeys to get the related
+        Project.
+        """
+        if isinstance(obj, Project):
+            return obj
+        relation = getattr(self, "project_relation", "project")
+        for part in relation.split("__"):
+            obj = getattr(obj, part)
+        return obj
```

Comment:
Can we have tests to validate that the mixin is able to retreive the relevant project for the different entities: 
- Feedback Form
- Responses
- Prompts
- PromptOptions

The tests can cover just this part of the logic, and nothing else though. 

This will be the foundation of the auth structure, and I'd like to be sure this work as expected.